x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcDatabaseMetaData.java

```diff
@@ -1258,10 +1259,10 @@ public boolean generatedKeyAlwaysReturned() throws SQLException {
                 EsType type = EsType.KEYWORD;
                 if (i + 1 < cols.length) {
                     Object next = cols[i + 1];
-                    // check if the next item it's a type
+                    // check if the next item is a type
                     if (next instanceof EsType || next instanceof JDBCType) {
                         try {
-                            type = TypeUtils.of((JDBCType) next);
+                            type = TypeUtils.of((SQLType) next);
```

Comment:
Casting an EsType to JDBCType triggers CCE, SQLType is the common type (and what TypeUtils#of takes).